src/misc/ethernet.cpp

```diff
@@ -0,0 +1,49 @@
+/*
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *
+ *  Copyright (C) 2021  The DOSBox Team
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+#include <cstring>
+
+#include "dosbox.h"
+#include "control.h"
+#include "ethernet.h"
+#include "ethernet_slirp.h"
+
+EthernetConnection *OpenEthernetConnection(const std::string &backend)
+{
+	EthernetConnection *conn = nullptr;
+	Section *settings = nullptr;
+#if C_SLIRP
+	if (backend == "slirp") {
+		conn = ((EthernetConnection *)new SlirpEthernetConnection);
+		settings = control->GetSection("ethernet, slirp");
+	}
+#endif
+	assert(settings);
+	if (!conn) {
+		LOG_MSG("ETHERNET: Unknown ethernet backend: %s", backend.c_str());
+		return nullptr;
+	}
```

Comment:
What do you propose I do if 'backend' isn't a valid backend then?

Reply:
But there's only 1 backend… Why do we need user-selectable option when we have a single backend? If NE2000 emulation is enabled - slirp will be used. If it's disabled - slirp won't be used.

Reply:
Do you want multiple backends?

Reply:
Perhaps to ask better: Why am I bothering with having an interface for Ethernet connections instead of having a single Slirp class that the NE2000 calls?

Reply:
> Do you want multiple backends?  Perhaps to ask better: Why am I bothering with having an interface for Ethernet connections instead of having a single Slirp class that the NE2000 calls?

 @dreamer's comment was against the error message that would be unnecessary if Ethernet & Slirp were toggled on/off together under a single `#define` value. 

@dreamer's comment did not put the back-end design into question; so I suggest leaving the back-end design as-is.  I agree that the interface might appear slightly over-engineered in our Slirp-only use-case, however, that code is in-common with DOSBox-X, and therefore if there are future changes it might be easier to address both code-bases in tandem.

Reply:
Because this is a programmer-side sanity check, replaced the `conn` check with an assert.